scene/gui/graph_edit.cpp

```diff
@@ -246,10 +246,25 @@ void GraphEdit::get_connection_list(List<Connection> *r_connections) const {
 
 void GraphEdit::set_scroll_ofs(const Vector2 &p_ofs) {
 	setting_scroll_ofs = true;
+	_set_scroll_ofs(p_ofs);
+	setting_scroll_ofs = false;
+}
+
+void GraphEdit::_set_scroll_ofs(const Vector2 &p_ofs) {
+	bool is_being_set = setting_scroll_ofs;
+	// _scroll_moved is called when each scroll value is set.
+	// So suppress the scroll_offset_changed signal until after they're set.
+	setting_scroll_ofs = true;
```

Comment:
This double use of the same flag is too convoluted and unreliable. I think the overall solution can be as simple as storing the the last offset and checking if it has changed before emitting the signal.